src/lightning/pytorch/trainer/setup.py

```diff
@@ -29,7 +28,12 @@
     XLAProfiler,
 )
 from lightning.pytorch.utilities.exceptions import MisconfigurationException
-from lightning.pytorch.utilities.imports import _HPU_AVAILABLE, _LIGHTNING_HABANA_AVAILABLE
+from lightning.pytorch.utilities.imports import (
+    _HPU_AVAILABLE,
+    _IPU_AVAILABLE,
```

Comment:
We should completely remove this from the repo and replace all usages of `_IPU_AVAILABLE` with `IPUAccelerator.is_available`

Reply:
but that you wont get the message that you are using IPU and just not installed related package, so I would keep it

Reply:
`IPUAccelerator.is_available` calls `_IPU_AVAILABLE` internally. My point is that we should only use the former

Reply:
yes but if you do not have `IPUAccelerator` you can call internal `_IPU_AVAILABLE`

Reply:
It's a staticmethod

Reply:
So lets leave it here for now and check it follow-up PR as this may also apply for HPU

Reply:
lets address it together with HPU in #17465

Reply:
I agree it's cleaner to use the function, IMO can be done in follow up